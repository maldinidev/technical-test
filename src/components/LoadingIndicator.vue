<template>
  <div v-if="loading" role="status" aria-live="polite" class="text-center py-4 text-gray-500">
    Cargando im√°genes...
  </div>
</template>

<script setup>
// Define the props for the component
defineProps({
  loading: {
    type: Boolean, // Indicates whether the loading state is active
    required: true, // This prop is mandatory for the component to render correctly
  },
});
</script>
