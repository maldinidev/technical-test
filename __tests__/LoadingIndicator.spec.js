import { mount } from '@vue/test-utils';
import LoadingIndicator from '../src/components/LoadingIndicator.vue';

describe('LoadingIndicator.vue', () => {
  it('muestra el mensaje de carga cuando `loading` es true', () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: true,
      },
    });

    expect(wrapper.text()).toContain('Cargando imágenes...');
  });

  it('no muestra el mensaje de carga cuando `loading` es false', () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: false,
      },
    });

    expect(wrapper.text()).not.toContain('Cargando imágenes...');
  });

  it('matches the snapshot when loading is true', () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches the snapshot when loading is false', () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the loading message only when `loading` is true', async () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: true,
      },
    });

    // Verify that the div is rendered
    expect(wrapper.find('div').exists()).toBe(true);

    // Change the loading prop to false
    await wrapper.setProps({ loading: false });

    // Verify that the div is removed
    expect(wrapper.find('div').exists()).toBe(false);
  });

  it('adds appropriate ARIA attributes when loading is true', () => {
    const wrapper = mount(LoadingIndicator, {
      props: {
        loading: true,
      },
    });

    const loadingDiv = wrapper.find('div');
    expect(loadingDiv.attributes('role')).toBe('status');
    expect(loadingDiv.attributes('aria-live')).toBe('polite');
  });
});
