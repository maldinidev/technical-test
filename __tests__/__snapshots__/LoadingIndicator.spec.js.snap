// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator.vue matches the snapshot when loading is false 1`] = `"<!--v-if-->"`;

exports[`LoadingIndicator.vue matches the snapshot when loading is true 1`] = `"<div role="status" aria-live="polite" class="text-center py-4 text-gray-500"> Cargando im√°genes... </div>"`;
